src/types/raw_request.ts

```diff
@@ -0,0 +1,31 @@
+import { EthCallParams } from "./eth_call";
+import { Uri } from "./shared";
+
+type _BaseRequest = {
+  cid: string;
+  jsps: string[];
+  pow: string | number;
+  time: number | string;
+  encoding: "json";
+};
+
+export type RawOracleHttpsRequest = {
+  uri: Uri;
```

Comment:
Let's add uri and encoding to the Base

Reply:
Please elaborate here. 
URI type is currently

```typescript
export type Uri = `http${string}` | `https${string}`;
```

This is simply to make it so that URIs for non ethApi requests adhere to the proper standard. 
Are you suggesting that it checks for valid encoding and if not runs encodeURI()?

If yes, can do something like this

```typescript
function getEncodedURI(uri: string) {
    if (decodeURI(uri) === uri) return uri;
    return encodeURI(uri);
}
```